From 5ef7aec9026b3c7dbca77005b5eaab5f25ac2f07 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Fri, 27 Sep 2013 15:46:51 +0800
Subject: [PATCH 27/52] ENGR00281564 [Android_ MX6Q\DL_SD\ARD] MPEG2Dec: No
 elapsed time

displayed and will pop up error at the end of playback if play one MPEG2
with Catus player. 100%

add protection when using mMediaPlayer in updateButtons()
do not use duration if it's 0 when checking current media position overflow

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
index e93f140..04fcfae 100755
--- a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
+++ b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
@@ -361,12 +361,12 @@ public class VideoPlayer extends Activity implements SeekBar.OnSeekBarChangeList
 
 			if(val1 > 0)
 				mDuration = val1;
-			else if(mDuration == 0)
+			else if(mDuration == 0 && mMediaPlayer != null)
 				mDuration = mMediaPlayer.getDuration();
 
 			val2 -= mTimeOffset; // offset by mTimeOffset
 
-			if(val2 > mDuration)
+			if(mDuration != 0 && val2 > mDuration)
 				val2 = mDuration;
 			else if(val2 < 0)
 				val2 = 0;
-- 
1.8.0

